ID2.0 in the browser wallet (#672)
Concordium/concordium.github.io: source/mainnet/net/resources/glossary.rst

will never be reset, and accounts created on the mainnet will remain
indefinitely.

.. _glossary-membership-proof:

Membership proof
================

> A proof to determine if a user has one or more attributes, for example, lives in the EU. Can also be a :ref:`non-membership proof<glossary-non-membership-proof>`.

Reviewer: ```suggestion
A proof to determine if an attribute of a users identity is included in a given set, for example, lives in the EU. Can also be a :ref:`non-membership proof<glossary-non-membership-proof>`.
```

   :ref:`epoch<glossary-epoch>` that is used to seed the :ref:`leader election<glossary-leader-election>` process.
-  :ref:`Transaction sequence number<glossary-transaction-sequence-number>` (same as account sequence number)

.. _glossary-non-membership-proof:

Non-membership proof
====================

> A proof to determine that a user does not meet one or more attributes, for example, that they are a resident of a country under trade sanctions.

Reviewer: ```suggestion
A proof to determine if an attribute of a users identity is **not** included in a set, for example, that they are **not** a resident of a country under trade sanctions.
```

Reviewer 2: I don't agree with this rephrasing. A proof of the non-membership statement determines that the user's attribute is not a member. Not whether they are a member. If the proof fails to verify then it does not mean that the attribute is not a member of the set. There could be other problems with the proof.

Reviewer: I don't get it. Either you are a member, or you're not a member? What falls inbetween or outside this in your opinion, and how would you phrase this?

Reviewer 2: My objection was to the phrasing "if". The only time a proof determines something is when it is valid. If it is not valid it does not mean that the statement is not true.

```
A proof to determine that an attribute of a user's identity is **not** included in a set, for example, that they are **not** a resident of a country under trade sanctions.
```

Reviewer: Ah, right.

A form of delegation where a delegator's stake is effectively distributed among all baker pools. It is not associated with a specific baker. Delegators earn lower rewards when delegating to passive delegation than when delegating to a specific baker pool. However, passive delegation is not affected by poor performance of a single baker.

.. _glossary-range-proofs:

Range proofs
============

> A range proof asks a user to prove that they meet an attribute within a range of values. For example, when renting a car, you might need to prove that you are between 25 and 65 to the car rental company. This could be constructed as a range proof.

Reviewer 2: ```suggestion
A range proof asks a user to prove that they meet an attribute within a range of values. For example, when renting a car, you might need to prove that you are between 25 and 65 years old to the car rental company. This could be constructed as a range proof.
```

A form of delegation where a delegator's stake is effectively distributed among all baker pools. It is not associated with a specific baker. Delegators earn lower rewards when delegating to passive delegation than when delegating to a specific baker pool. However, passive delegation is not affected by poor performance of a single baker.

.. _glossary-range-proofs:

Range proofs
============

A range proof asks a user to prove that they meet an attribute within a range of values. For example, when renting a car, you might need to prove that you are between 25 and 65 years old to the car rental company. This could be constructed as a range proof.

.. _glossary-reveal-attribute:

Reveal
======

> To reveal an attribute. This can be used in identity verification proof. When you reveal an attribute, you give the dApp or service that requested it your exact information, such as date of birth, or nationality. You should only do this if you have **absolute trust** in them, and if you are familiar with their data usage and protection procedures.

Author: It was already done. Think you need to pull the latest. I did that yesterday after comment about it from Ales.

:ref:`Delegators<glossary-delegate>` can delegate stake to a baker pool or passive delegation. This affects the staked amount of the baker and thus their lottery power.

.. _glossary-statement:

Statement
=========

> One or more attributes to be proven or revealed by a wallet owner to a dApp or service.

Reviewer 2: The part about "to be proven" does not read very well.

Maybe the following is better?
"A list whose items are either attributes to reveal, or properties of attributes to prove."

The winning probability is the probability that a baker wins in a given slot.
The probability is :math:`1-(1-f)^α`, where :math:`f` is the difficulty parameter and :math:`α` is
the :ref:`lottery power<glossary-lottery-power>`.

.. _glossary-zero-knowledge-proof:

Zero-knowledge proof
====================

> A method by which a user (the prover) can prove to another party (the verifier) that the user meets a requirement without revealing anything beyond the truth of the proven statement.

Reviewer 2: This mixes two ways of stating things. "requirement" and "proven statement".

Perhaps this would be better.
```
A method by which a user (the prover) can prove to another party (the verifier) that the user meets a requirement without revealing anything beyond that.
```
